### PR TITLE
Feature/je 101 expenses

### PR DESCRIPTION
**Pull request by rimi-itk**

### CODE REVIEW

File: bundles/Expense/Controller/Admin/DefaultController.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * This source file is subject to the MIT license.
+ */
+
+namespace Expense\Controller\Admin;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * Class ExpenseController.
```
Comment (tuj) [inline bundles/Expense/Controller/Admin/DefaultController.php]: ExpenseController or DefaultController?

File: bundles/Expense/Controller/Admin/ExpenseCategoryController.php [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+use Expense\Repository\ExpenseCategoryRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Class ExpenseController.
```
Comment (tuj) [inline bundles/Expense/Controller/Admin/ExpenseCategoryController.php]: ExpenseController or ExpenseCategoryController?

File: bundles/Expense/Controller/ExpenseController.php [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        if (!empty($result->sections)) {
+            $sections = array_filter($result->sections, function ($section) {
+                return !empty($section->issues);
+            });
+            foreach ($sections as $section) {
+                if ('cs' === $section->id) {
+                    return new JsonResponse($section);
+                }
+            }
+//            return new JsonResponse($result->sections[0]);
```
Comment (tuj) [inline bundles/Expense/Controller/ExpenseController.php]: Remove

File: bundles/Expense/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+class Configuration implements ConfigurationInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->getRootNode('expense');
```
Comment (tuj) [inline bundles/Expense/DependencyInjection/Configuration.php]: RootNode name should be set as parameter to TreeBuilder constructor call.

$treeBuilder = new TreeBuilder('expense');

File: bundles/Expense/Repository/ExpenseCategoryRepository.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ * @method Category[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class ExpenseCategoryRepository extends ServiceEntityRepository
+{
+    public function __construct(RegistryInterface $registry)
+    {
+        parent::__construct($registry, Category::class);
+    }
+
+    // /**
```
Comment (tuj) [inline bundles/Expense/Repository/ExpenseCategoryRepository.php]: Remove

File: bundles/Expense/Resources/assets/index.js [hunk 6]
```diff
@@ -0,0 +1 @@
+require('./new.js');
```
Comment (tuj) [inline bundles/Expense/Resources/assets/index.js:1]: What is the point of this file?

File: bundles/Expense/Resources/assets/new.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+require('./scss/new.scss')
+
+$(() => {
+    const $issueCtrl = $('#form_issue_key').hide()
+    const $issuePicker = $('<select class="form-control"></select>')
+        .insertBefore($issueCtrl)
+        .css({width: '100%'})
+        .select2({
+            // placeholder: Expense.messages.select_project
```
Comment (tuj) [inline bundles/Expense/Resources/assets/new.js]: Remove?

File: bundles/Expense/Resources/translations/messages.da.xlf [hunk 8]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (tuj) [inline bundles/Expense/Resources/translations/messages.da.xlf]: Why not yaml?

File: bundles/Expense/Resources/views/admin/expense/category/_delete_form.html.twig [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<form method="post" action="{{ path('expense_admin_category_delete', {'id': category.id}) }}" onsubmit="return confirm('Are you sure you want to delete this item?');">
```
Comment (tuj) [inline bundles/Expense/Resources/views/admin/expense/category/_delete_form.html.twig]: English...

File: bundles/Expense/Resources/views/admin/expense/category/_form.html.twig [hunk 10]
```diff
@@ -0,0 +1,4 @@
+{{ form_start(form) }}
+    {{ form_widget(form) }}
+    <button class="btn btn-success">{{ button_label|default('Save') }}</button>
```
Comment (tuj) [inline bundles/Expense/Resources/views/admin/expense/category/_form.html.twig]: translate?

File: bundles/Expense/Resources/views/admin/expense/category/new.html.twig [hunk 11]
```diff
@@ -0,0 +1,11 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}New Category{% endblock %}
+
+{% block content %}
+    <h1>Create new Category</h1>
+
+    {{ include('@ExpenseBundle/admin/expense/category/_form.html.twig') }}
+
+    <a class="btn btn-link" href="{{ path('expense_admin_category_index') }}">back to list</a>
```
Comment (tuj) [inline bundles/Expense/Resources/views/admin/expense/category/new.html.twig]: Translate

File: bundles/Expense/Resources/views/admin/expense/category/show.html.twig [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+                <td>{{ category.id }}</td>
+            </tr>
+            <tr>
+                <th>Name</th>
+                <td>{{ category.name }}</td>
+            </tr>
+        </tbody>
+    </table>
+
+    <a class="btn btn-link" href="{{ path('expense_admin_category_index') }}">back to list</a>
```
Comment (tuj) [inline bundles/Expense/Resources/views/admin/expense/category/show.html.twig]: Translate

File: bundles/Expense/Resources/views/admin/index.html.twig [hunk 13]
```diff
@@ -0,0 +1,5 @@
+{% extends 'base.html.twig' %}
+
+{% block content %}
+    <a href="{{ path('expense_admin_category_index') }}">{{ 'Expense categories' }}</a>
```
Comment (tuj) [inline bundles/Expense/Resources/views/admin/index.html.twig]: Translate?

File: src/Service/AbstractJiraService.php [hunk 14]
```diff
@@ -72,6 +74,7 @@ public function post($path, $data)
                 return json_decode($body);
             }
         } catch (RequestException $e) {
+//            header('Content-type: text/plain'); echo var_export((string)$e->getResponse()->getBody(), true); die(__FILE__.':'.__LINE__.':'.__METHOD__);
```
Comment (tuj) [inline src/Service/AbstractJiraService.php]: Remove

File: src/Service/AbstractJiraService.php [hunk 15]
```diff
@@ -104,6 +107,29 @@ public function put($path, $data)
...
+     *
+     * @param $path
+     *
+     * @return mixed
+     */
+    protected function delete($path)
+    {
+        $client = $this->getClient();
+
+        // Set the "auth" request option to "oauth" to sign using oauth
```
Comment (tuj) [inline src/Service/AbstractJiraService.php]: Comment does not seem related to code below

File: templates/main/index.html.twig [hunk 16]
```diff
@@ -4,7 +4,7 @@
     ordre: { title: "Ordre", desc: "Opret ordrepakker ud fra tasks i Jira", icon: "fa-box-open", path: "/", btnTxt: "Pak en Ordre" },
     sprintplan: { title: "Sprintplan", desc: "Planlægningsoversigt og sprintrapporter baseret på tasks i Jira", icon: "fa-braille", path: "/", btnTxt: "Se Sprintplan" },
     nytProjekt: { title: "Nyt projekt", desc: "Opret et nyt Jira-projekt ud fra en skabelon", icon: "fa-project-diagram", path: "/create-project/new", btnTxt: "Opret Nyt projekt" },
-    udgift: { title: "Udgift", desc: "Opret udgifter i forbindelse med projekter og tasks", icon: "fa-credit-card", path: "/", btnTxt: "Tilføj en Udgift" },
+    udgift: { title: "Udgift", desc: "Opret udgifter i forbindelse med projekter og tasks", icon: "fa-credit-card", path: "/expense/new", btnTxt: "Tilføj en Udgift" },
```
Comment (tuj) [inline templates/main/index.html.twig]: Add to Controller instead that will be added by https://github.com/aakb/jira_economics/pull/14

File: templates/navigation/global-nav.html.twig [hunk 17]
```diff
@@ -5,7 +5,7 @@
     ordre: { title: "Ordre", desc: "Opret ordrepakker ud fra tasks i Jira", icon: "fa-box-open", path: "/", active: false },
     sprintplan: { title: "Sprintplan", desc: "Planlægningsoversigt og sprintrapporter baseret på tasks i Jira", icon: "fa-braille", path: "/", active: false },
     nytProjekt: { title: "Nyt projekt", desc: "Opret et nyt Jira-projekt ud fra en skabelon", icon: "fa-project-diagram", path: "/create-project/new", active: false },
-    udgift: { title: "Udgift", desc: "Opret udgifter i forbindelse med projekter og tasks", icon: "fa-credit-card", path: "/", active: false },
+    udgift: { title: "Udgift", desc: "Opret udgifter i forbindelse med projekter og tasks", icon: "fa-credit-card", path: "/expense/new", active: false },
```
Comment (tuj) [inline templates/navigation/global-nav.html.twig]: Add to MenuService from https://github.com/aakb/jira_economics/pull/14

**Review by tuj — CHANGES_REQUESTED** [review]

File: bundles/Expense/Resources/assets/index.js (hunk 6 above)
Comment (rimi-itk) [reply bundles/Expense/Resources/assets/index.js:1]: Consistency with (most) other bundles.

File: bundles/Expense/Resources/translations/messages.da.xlf (hunk 8 above)
Comment (rimi-itk) [reply bundles/Expense/Resources/translations/messages.da.xlf]: “translations can be created in several different formats, XLIFF being the recommended format” [https://symfony.com/doc/current/translation.html#basic-translation]

It was mostly an experiment – I'll switch to `yaml`.

**Review by tuj — APPROVED** [review]
